src/spec.c

```diff
@@ -2246,7 +2334,15 @@ static void IndexSpec_ScanAndReindexAsync(StrongRef spec_ref) {
 #ifdef _DEBUG
   RedisModule_Log(RSDummyContext, "notice", "Register index %s for async scan", ((IndexSpec*)StrongRef_Get(spec_ref))->name);
 #endif
-  IndexesScanner *scanner = IndexesScanner_New(spec_ref);
+  IndexesScanner *scanner = NULL;
+  if (debugCtx.debugMode) {
+    scanner = (IndexesScanner*)DebugIndexesScanner_New(spec_ref);
+    IndexSpec *sp = StrongRef_Get(spec_ref);
+    sp->flags |= Index_DebugScanner;
```

Comment:
Does that mean the current scanner is a debug scanner?

1. What happens if the current scanner is being canceled?
2. When do we turn off this flag?

Reply:
If we start in scan while in debugMode we assume we need a debug scanner.
Each Spec is assigned a  new scanner (once), if it's canceled then the scan is finished, and whatever not indexed so far will not be indexed. The scanner object is freed later on.
So the flag should always stay up, as we don't create a another scanner for that index.
Does this makes sense?

Reply:
Debug scanner flag was moved into IndexesScanner base class.